src/compat/posix/pwd_grp/pwd_grp.c

```diff
@@ -73,8 +75,10 @@ static int read_int_field(FILE *stream, const char *format, void *field, int del
 	sscanf("0", format, field);
 
 	ret = fscanf(stream, format, field);
-	if (0 > ret && ferror(stream)) {
-		return -ret;
+	if (0 > ret) {
+		if (ferror(stream))
+			return EIO;
+		return EOF;
```

Comment:
Thanks for your contribution. We don't merge since ssh test is broken on Travis. First thing is this line - when `fscanf` returns something lower than 0, it probably means it's empty field , that is `::`. So we can't just return `EOF` here, it breaks `fgetspent` for example.

Reply:
Probably, we shall delete `return EOF` and either leave a comment about empty fields `::` and fall through below. As I can see next lines below are checking something about field emptiness already.

Reply:
Done